purchase_order_approval_block/models/purchase_order.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    approval_block_id = fields.Many2one(
+        comodel_name='purchase.approval.block.reason',
+        string='Approval Block Reason',
+    )
+    approval_blocked = fields.Boolean('Approval Blocked',
```

Comment:
Why again a boolean? I don't get it, is it needed for another purpose? In this module seems useless.

Reply:
Probable it's there for technical purposes. It's used in view's domains

Reply:
I see, to not be able to change ever the block reason once it is set? seems reasonble. Is that the desired behavior?

Reply:
yes, that's it.